plugin/storage/cassandra/spanstore/reader.go

```diff
@@ -322,7 +336,7 @@ func (s *SpanReader) queryByTagsAndLogs(ctx context.Context, tq *spanstore.Trace
 			tq.NumTraces*limitMultiple,
 		).PageSize(0)
 		t, err := s.executeQuery(childSpan, query, s.metrics.queryTagIndex)
-		childSpan.Finish()
+		defer childSpan.End()
```

Comment:
```suggestion
		childSpan.End()
```
the span is a part of an inner loop, it needs to be finished before the next iteration